src/connector/src/sink/clickhouse.rs

```diff
@@ -0,0 +1,629 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::fmt::Debug;
+use std::collections::HashMap;
+
+use anyhow::anyhow;
+use bytes::{BufMut, BytesMut};
+use clickhouse::update::Field;
+use clickhouse::{Client, Row as ClickHouseRow};
+use itertools::Itertools;
+use risingwave_common::array::{Op, RowRef, StreamChunk};
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::catalog::Schema;
+use risingwave_common::row::Row;
+use risingwave_common::types::{DatumRef, ScalarRefImpl, ToText};
+use risingwave_common::util::iter_util::ZipEqFast;
+use risingwave_rpc_client::ConnectorClient;
+use serde_derive::Deserialize;
+use serde_with::serde_as;
+
+use super::{DummySinkCommitCoordinator, SinkWriterParam};
+use crate::common::ClickHouseCommon;
+use crate::sink::{
+    Result, Sink, SinkError, SinkWriter, SINK_TYPE_APPEND_ONLY, SINK_TYPE_DEBEZIUM,
+    SINK_TYPE_OPTION, SINK_TYPE_UPSERT,
+};
+use crate::source::DataType;
+
+pub const CLICKHOUSE_SINK: &str = "clickhouse";
+const BUFFER_SIZE: usize = 1024;
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+pub struct ClickHouseConfig {
+    #[serde(flatten)]
+    pub common: ClickHouseCommon,
+
+    pub r#type: String, // accept "append-only" or "upsert"
```

Comment:
`type` is a built-in option key in kernel. Therefore, I don't think we need to have a field here, or we only parse it but do not read it. 

The `type` specified in the with props are handled and put in the `SinkParam::sink_type`. Please use the `sink_type` to determine whether a sink is append-only or upsert.

Reply:
If we set type to ‘debezium’, ‘data_type’ is set to ‘upsert’. This is not consistent with ‘r#type’ and may not be allowed. Also, shouldn't the parameters of the method ‘from_hashmap’ be consistent with the other sink